zetaclient/chains/bitcoin/observer/gas_price.go

```diff
@@ -0,0 +1,70 @@
+package observer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/zeta-chain/node/pkg/chains"
+	"github.com/zeta-chain/node/zetaclient/chains/bitcoin/client"
+	"github.com/zeta-chain/node/zetaclient/chains/bitcoin/common"
+)
+
+// PostGasPrice posts gas price to zetacore
+func (ob *Observer) PostGasPrice(ctx context.Context) error {
+	var (
+		err              error
+		feeRateEstimated int64
+	)
+
+	// special handle regnet and testnet gas rate
+	// regnet:  RPC 'EstimateSmartFee' is not available
+	// testnet: RPC 'EstimateSmartFee' returns unreasonable high gas rate
+	if ob.Chain().NetworkType != chains.NetworkType_mainnet {
+		feeRateEstimated, err = ob.specialHandleFeeRate(ctx)
+		if err != nil {
+			return errors.Wrap(err, "unable to execute specialHandleFeeRate")
+		}
+	} else {
+		isRegnet := chains.IsBitcoinRegnet(ob.Chain().ChainId)
```

Comment:
This is always false if we filter above ob.Chain().NetworkType != chains.NetworkType_mainnet ??

Reply:
simplified as `false` https://github.com/zeta-chain/node/pull/3381/commits/37f4ebbd5ed1c86d7ff7be5912c7c0cf7212a0d6